### PR TITLE
Add smoke test for Logback MDC instrumentation

### LINKED ISSUE #1209: Smoke test for Logback MDC instrumentation
The logback instrumentation is fragile (#1201) because logback depends on slf4j which we shade during our build, and our normal tests don't currently catch shading issues.

We are hoping to change how we run tests post-GA, so that they run using a real (and shaded) javaagent, which would catch these issues.

For now though, it maybe worth adding a smoke test to make sure this doesn't accidentally break again without us knowing about it.

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Resolves #1209

Needs to be merged after changes from https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1228 are deployed

### CODE REVIEW

File: smoke-tests/springboot/src/main/java/io/opentelemetry/smoketest/springboot/controller/WebController.java [hunk 1]
```diff
@@ -17,13 +17,18 @@
...
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 public class WebController {
+  private static final Logger LOGGER = LoggerFactory.getLogger(WebController.class);
+
   @RequestMapping("/greeting")
   public String greeting() {
+    LOGGER.info("HTTP request received");
```
Comment (iNikem) [inline smoke-tests/springboot/src/main/java/io/opentelemetry/smoketest/springboot/controller/WebController.java:31]: Tests fail because they don't build this application as part of the test. If you are sure that test work, you may submit two separate PRs: one for this app and another for the actual test

Comment (mateuszrzeszutek) [reply smoke-tests/springboot/src/main/java/io/opentelemetry/smoketest/springboot/controller/WebController.java:31]: I've tested it locally and it worked -- split into two PRs.
Thanks!

**Comment by iNikem** [comment]
please rebase

File: smoke-tests/src/test/groovy/io/opentelemetry/smoketest/SmokeTest.groovy [hunk 2]
```diff
@@ -172,4 +180,15 @@ abstract class SmokeTest extends Specification {
 
     return content
   }
+
+  protected static Set<String> getLoggedTraceIds(ToStringConsumer output) {
+    output.toUtf8String().lines()
+      .flatMap(SmokeTest.&findTraceId)
```
Comment (trask) [inline smoke-tests/src/test/groovy/io/opentelemetry/smoketest/SmokeTest.groovy:186]: oh groovy 😭

**Review by trask — APPROVED** [review]
